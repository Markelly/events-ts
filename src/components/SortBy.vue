<template>
  <div class="sort-by ml-auto">
    <b-form-select
        v-model="selected"
        :options="options"
        @change="sort(selected)"
    />
  </div>
</template>

<script lang="ts">
import { SORT_OPTIONS } from "@/utils/constants";
import { Component, Vue } from "vue-property-decorator";

@Component
export default class SortBy extends Vue {
  selected: string = "";
  options: object[] = SORT_OPTIONS;

  sort(selected: string) {
    this.$emit("sort", selected);
  }
};
</script>

<style scoped lang="scss">
@import "@/styles/events-ts";

.sort-by {
  padding-top: 8px;
  color: $color-secondary;

  @media (min-width: $breakpoint-tablet) {
    width: 240px;
  }
}
</style>
