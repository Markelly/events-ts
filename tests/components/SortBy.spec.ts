import SortBy from '@/components/SortBy.vue';
import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('<SortBy/>', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(SortBy, {
      localVue
    });
  });

  test('should render SortBy component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should emit sortEvents event', () => {
    wrapper.vm.sort('date,asc');
    expect(wrapper.emitted('sort', 'date,asc')).toBeTruthy();
  });
});
