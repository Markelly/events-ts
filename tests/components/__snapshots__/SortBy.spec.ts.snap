// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortBy/> should render SortBy component 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`<SortBy/> should render the component 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
